spec/controllers/challenge_assignments_controller_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'spec_helper'
+
+describe ChallengeAssignmentsController do
+  include LoginMacros
+  include RedirectExpectationHelper
+
+  context "when not logged in" do
+    describe 'no_challenge' do
+      before(:each) do
+        @collection = FactoryGirl.create(:collection)
```

Comment:
I think you'd probably be okay doing `let(:collection) { create(:collection) }` at the top instead of using before blocks to create instance variables. But either way, you can omit `FactoryGirl` for brevity. 

The "should" can also be left out of descriptions -- it's less repetitive and lets you get right to the important words.